pygmt/x2sys.py

```diff
@@ -263,8 +286,23 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
             if kind == "file":
                 file_contexts.append(dummy_context(track))
             elif kind == "matrix":
-                raise NotImplementedError(f"{type(track)} inputs are not supported yet")
-                # file_contexts.append(lib.virtualfile_from_matrix(track.values))
+                # find suffix (-E) of trackfiles used (e.g. xyz, csv, etc) from
+                # $X2SYS_HOME/TAGNAME/TAGNAME.tag file using regex search
+                lastline = (
+                    Path(os.environ["X2SYS_HOME"], kwargs["T"], f"{kwargs['T']}.tag")
+                    .read_text()
+                    .strip()
+                    .split("\n")[-1]
+                )  # e.g. "-Dxyz -Etsv -I1/1"
+                try:
+                    # 1st try to match file extension after -E
+                    suffix = re.search(pattern=r"-E(\S*)", string=lastline).group(1)
+                except AttributeError:  # 'NoneType' object has no attribute 'group'
+                    # 2nd try to match file extension after -D
+                    suffix = re.search(pattern=r"-D(\S*)", string=lastline).group(1)
```

Comment:
Is there a better way to check if `-Exyz` is in the string, and if not, fallback to parsing from `Dxyz`?

Reply:
What about this one?
```python
lastline = "-Dxyz -Etsv -I1/1"
#lastline = "-Dxyz -I1/1"

for item in lastline.split():
    for key in ['-E', '-D']:
        if item.startswith(key):
            suffix = item[2:]
            break
print(suffix)
```

Reply:
Hmm, this gives me some ideas. I'll play around with it, thanks!